src/com/esotericsoftware/yamlbeans/YamlReader.java

```diff
@@ -1,424 +1,435 @@
-		public YamlReaderException (String message, Throwable cause) {
-			super("Line " + parser.getLineNumber() + ", column " + parser.getColumn() + ": " + message, cause);
-		}
-
-		public YamlReaderException (String message) {
-			this(message, null);
-		}
-	}
-
-	public static void main (String[] args) throws Exception {
-		YamlReader reader = new YamlReader(new FileReader("test/test.yml"));
-		System.out.println(reader.read());
-	}
-}
+/*
+ * Copyright (c) 2008 Nathan Sweet
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software
+ * is furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+ * IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.esotericsoftware.yamlbeans;
+
+import static com.esotericsoftware.yamlbeans.parser.EventType.*;
+
+import com.esotericsoftware.yamlbeans.Beans.Property;
+import com.esotericsoftware.yamlbeans.parser.AliasEvent;
+import com.esotericsoftware.yamlbeans.parser.CollectionStartEvent;
+import com.esotericsoftware.yamlbeans.parser.Event;
+import com.esotericsoftware.yamlbeans.parser.Parser;
+import com.esotericsoftware.yamlbeans.parser.Parser.ParserException;
+import com.esotericsoftware.yamlbeans.parser.ScalarEvent;
+import com.esotericsoftware.yamlbeans.scalar.ScalarSerializer;
+import com.esotericsoftware.yamlbeans.tokenizer.Tokenizer.TokenizerException;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/** Deserializes Java objects from YAML.
+ * @author <a href="mailto:misc@n4te.com">Nathan Sweet</a> */
+public class YamlReader {
+	private final YamlConfig config;
+	Parser parser;
+	boolean allowDuplicates = true;
+	private final Map<String, Object> anchors = new HashMap();
+
+	public YamlReader (Reader reader) {
+		this(reader, new YamlConfig());
+	}
+
+	public YamlReader (Reader reader, boolean allowDuplicates) {
```

Comment:
Don't add a constructor, the setting should go in YamlConfig like all other settings.